x/payment/spec/docs.md

```diff
@@ -1,117 +1,46 @@
-# Abstract
+# `x/payment`
```

Comment:
A follow-up PR will move this file to `x/payment/README.md` per https://github.com/celestiaorg/celestia-app/issues/330#issuecomment-1303118116

Reply:
https://github.com/celestiaorg/celestia-app/pull/1057